### PR TITLE
add command: actor get by address

### PR DESCRIPTION
**Pull request by GXKe**
#3430

### CODE REVIEW

File: commands/actor.go [hunk 1]
```diff
@@ -107,6 +110,62 @@ var actorLsCmd = &cmds.Command{
...
+		result, err := GetPorcelainAPI(env).ActorGet(req.Context, addr)
+		if err != nil {
+			return err
+		}
+
+		addrStr := addr.String()
+		var output *ActorView
+		switch {
+		case result.Empty(): // empty (balance only) actors have no Code.
+			output = makeActorView(result, addrStr, nil)
```
Comment (ingar) [inline commands/actor.go]: As this code is essentially the same as the code for `actor ls`, let's move it to a private helper for forward-looking maintainability.

File: commands/actor.go [hunk 2]
```diff
@@ -107,6 +110,62 @@ var actorLsCmd = &cmds.Command{
...
+		}
+		if err := re.Emit(output); err != nil {
+			return err
+		}
+		return nil
+	},
+	Type: &ActorView{},
+	Encoders: cmds.EncoderMap{
+		cmds.JSON: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, a *ActorView) error {
```
Comment (ingar) [inline commands/actor.go]: Same thing here for this encoder, as it's the same as for `actor ls`

File: commands/actor.go (hunk 1 above)
Comment (GXKe) [reply commands/actor.go]: go-filecoin actor ls --address <actor address>
good idea

**Comment by GXKe** [comment]
> Thank you for the PR.
> 
> Couple of things:
> 
> * I have a couple of comments about reducing the duplication in the file.
> * If there ends up being two actor commands, `ls` and `get`, some `Helptext` would be nice for each command.
> * Along those lines, this could be written with an argument to `ls`, as in `go-filecoin actor ls --address <actor address>`.  I'm unclear if it would be cleaner or not.
> * Tests need to be added to `commands/actor_daemon_test.go`

Has been resolved

**Comment by frrist** [comment]
Thanks @GXKe! @ingar and @ZenGround0 what do you think of putting this on the `show` command? It would then look similar to:
```
USAGE
  go-filecoin show - Get human-readable representations of filecoin objects

SYNOPSIS
  go-filecoin show

SUBCOMMANDS
  go-filecoin show block <cid>    - Show a full filecoin block by its header CID
  go-filecoin show header <cid>   - Show a filecoin block header by its CID
  go-filecoin show messages <cid> - Show a filecoin message collection by its CID
  go-filecoin show receipts <cid> - Show a filecoin receipt collection by its CID
  go-filecoin show actor <address> - Show a filecoin actor by its Address.
```

File: commands/actor.go [hunk 3]
```diff
@@ -52,10 +55,35 @@ var actorCmd = &cmds.Command{
...
-		if err != nil {
-			return err
+		addrString, _ := req.Options["address"].(string)
+		var results <-chan state.GetAllActorsResult
+		if len(addrString) > 0 {
+			addr, err := address.NewFromString(addrString)
+			if err != nil {
+				return err
+			}
+			act, err := GetPorcelainAPI(env).ActorGet(req.Context, addr)
```
Comment (frrist) [inline commands/actor.go]: My preference would be to make this a separate command, either on `actor` or `show` as stated elsewhere. I don't think we need to be using a channel for this and adding it as a separate command would clean up that logic.

**Comment by anorth** [comment]
`show` should be disbanded as a top-level command. Don't put more stuff under it. See #1587

**Comment by ZenGround0** [comment]
This may be worth using in the future after scoping CLI redesign work and checking that it is in scope.  Closing for now.

**Pull request closed without merging**